lgc/patch/PatchLoopMetadata.cpp

```diff
@@ -70,6 +76,38 @@ class PatchLoopMetadata : public Patch {
 // Initializes static members.
 char PatchLoopMetadata::ID = 0;
 
+// =====================================================================================================================
+// Update metadata by removing any existing metadata with the specified prefix, and then adding the new metadata if
+// existing metadata was removed or conditional is false.
+MDNode *PatchLoopMetadata::updateMetadata(MDNode *LoopID, ArrayRef<StringRef> RemovePrefixes, Metadata *newMetadata,
+                                          bool conditional) {
+  bool found = false;
+  SmallVector<Metadata *, 4> MDs;
+  // Reserve first location for self reference to the LoopID metadata node.
+  TempMDTuple tempNode = MDNode::getTemporary(*m_context, None);
+  MDs.push_back(tempNode.get());
+  for (unsigned i = 1, e = LoopID->getNumOperands(); i < e; ++i) {
+    Metadata *op = LoopID->getOperand(i);
+    if (MDNode *mdNode = dyn_cast<MDNode>(op)) {
+      const MDString *mdString = dyn_cast<MDString>(mdNode->getOperand(0));
+      if (mdString)
```

Comment:
nit 1: can you move the definition of mdString to the if condition? (same like above)
nit 2: can you add some more braces around this if? The if body contains another if-else,

Reply:
done